cfme/services/catalogs/catalog_item.py

```diff
@@ -140,273 +213,276 @@ def create(self):
-        if not (self.item_type in provider_required_types):
-            sel.click(basic_info_form.field_entry_point)
+        view.fill({'name': self.name,
+                   'description': self.description,
+                   'display': self.display_in,
+                   'select_catalog': self.catalog.name,
+                   'select_dialog': self.dialog,
+                   'select_orch_template': orch_template,
+                   'select_provider': provider_name,
+                   'select_config_template': self.config_template})
+
+        if view.field_entry_point.value == "":
+            view.fill({'field_entry_point': 'a'})
             if version.current_version() < "5.7":
-                dynamic_tree.click_path("Datastore", self.domain, "Service", "Provisioning",
-                                     "StateMachines", "ServiceProvision_Template", "default")
+                view.dynamic_tree.click_path("Datastore", self.domain, "Service", "Provisioning",
+                    "StateMachines", "ServiceProvision_Template", "CatalogItemInitialization")
             else:
-                entry_tree.click_path("Datastore", self.domain, "Service", "Provisioning",
-                    "StateMachines", "ServiceProvision_Template", "default")
-            sel.click(basic_info_form.apply_btn)
+                view.entry_tree.click_path("Datastore", self.domain, "Service", "Provisioning",
+                    "StateMachines", "ServiceProvision_Template", "CatalogItemInitialization")
+            view.apply_button.click()
         if version.current_version() >= "5.7" and self.item_type == "AnsibleTower":
-            sel.click(basic_info_form.retirement_entry_point)
-            entry_tree.click_path("Datastore", self.domain, "Service", "Retirement",
+            view.fill({'retirement_entry_point': ''})
+            view.entry_tree.click_path("Datastore", self.domain, "Service", "Retirement",
                     "StateMachines", "ServiceRetirement", "Generic")
-            sel.click(basic_info_form.apply_btn)
+            view.apply_button.click()
+
         if self.catalog_name is not None \
                 and self.provisioning_data is not None \
                 and not isinstance(self.provider, NoneType):
             tabstrip.select_tab("Request Info")
             tabstrip.select_tab("Catalog")
-            template = template_select_form.template_table.find_row_by_cells({
-                'Name': self.catalog_name,
-                'Provider': self.provider.name
-            })
-            sel.click(template)
+            row = view.template_table.row(name=self.catalog_name, provider=self.provider.name)
+            row.click()
             request_form.fill(self.provisioning_data)
-        sel.click(template_select_form.add_button)
+        view.add_button.click()
+        view.flash.assert_success_message('Service Catalog Item "{}" was added'.format(self.name))
```

Comment:
Cannot put flash below view because of a test where error is expected in flash message .
If the flash is before error is caught else it fails at view assert